### PR TITLE
🎨 Reusable-image-and-text-banner-#8

### LINKED ISSUE #8: 🎨 Create Reusable Image and Text Banner
#  Create reusable text and image banner (side by side)
---

#### Note: This issue is dependent on #19 and #27 , please complete that before you continue.

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a component I can reuse throughout the site to display images and text side by side.

### Description

This component displays text and images side by side on the site

text image component

text image component2

Your assignment is to implement the component to appear as above.

Please take note of all the design features (including mobile and rounded corners)

#### Link to full design:
```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/
```
### Technical Analysis

This component will be created in `libs/elements/banners` (banners is a module). The component will be called elewa-image-and-text-banner. The component reuses the image-container and is therefore dependant on #19 . The component reuses the title and paragraph component in #27.

The section has an input of the below type to the component that will be passed down to the reusable components in the linked issue:

ImageConfig -> #19 and ContentText -> #27

```typeScript
interface ImageAndText {
  content: ContentText
  image: ImageConfig
  imagePosition: 'left' | 'right'
}
```

### Acceptance Criteria

1. When rendered the component is pixel-perfect compared to the design above.
2. The component is clean and does not have more than 50 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
5. The component is aligned using CSS flex
6. The design is responsive for screen, tablet and phone
7. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by philipNgungi037**
# Description

Creates a reusable image and text banner. 

Fixes #8

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Beulah-Matt — CHANGES_REQUESTED** [review]
Hi @philipNgungi037 take note of the issue scope. 
This component is supposed to be in libs/elements/banners. 

Move the files there then tag me when the changes are done.

**Comment by philipNgungi037** [comment]
The changes noted and I have moved the files.
@Beulah-Matt

**Review by Beulah-Matt — APPROVED** [review]
Good job. Changes approved.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Reuse the reusable components specified. Other than that everything else is clean

**Comment by philipNgungi037** [comment]
Changes submitted @LemmyMwaura .

File: libs/elements/banners/src/lib/elewa-image-and-text-banner/elewa-image-and-text-banner.component.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      33.3vw"
+      />
+    </div>
+
+    <!-- Container for the text content -->
+    <div class="text-container">
+      <h2>{{ ImageAndText.content.title }}</h2>
+      
+      <!-- Angular *ngFor directive to iterate through and display multiple descriptions -->
+      <p *ngFor="let description of ImageAndText.content.descriptions">{{ description }}</p>
```
Comment (LemmyMwaura) [inline libs/elements/banners/src/lib/elewa-image-and-text-banner/elewa-image-and-text-banner.component.html:21]: Re-use the components specified in the issue to render this. Render and pass inputs to them as it's been done in other pages.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just one small change left

File: libs/elements/banners/src/lib/elewa-image-and-text-banner/elewa-image-and-text-banner.component.html (hunk 1 above)
Comment (philipNgungi037) [reply libs/elements/banners/src/lib/elewa-image-and-text-banner/elewa-image-and-text-banner.component.html]: Done @LemmyMwaura

**Review by LemmyMwaura — DISMISSED** [review]

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**